src/packed-types/PackingPlant.ts

```diff
@@ -0,0 +1,79 @@
+import { Bool, Character, Field, Struct, UInt32, provable } from 'snarkyjs';
+
+type SupportedType = Bool | UInt32 | Character;
+
+const bitSizes = {
+  Bool: 1n,
+  UInt32: 32n,
+  Character: 16n,
+};
+
+export function PackingPlant<T>(elementType: T, l: number) {
```

Comment:
Ideally I could use the packing plant to initialize the other classes, or replace them entirely.  Right now this file does nothing.

I was struggling to get typescript to be happy with this.  For instance, if a function returns the type `Array[T]`, and my implementation returns `Array[Bool]`, it will tell me

```
Type 'Bool[]' is not assignable to type '(typeof Bool & ((x: boolean | Bool | FieldVar) => Bool))[]'.
```

Reply:
If I could solve this other point, then I think the Packing Plant could be a single-function implementation for any supported type:

- To pack and unpack an auxiliary type, we need to be able to convert the type to a Field, and a BigInt to the type
  - With Bool and UInt32, the names of the methods to do this are different, so the logic can't be put in a single file.

My idea is to have the supported classes that implement `pack` and `unpack` all be initialized by calling the Packing Plant, which is possible after solving this type error.  But even better would be to skip the other files altogether and just have the main file know how to pack and unpack whatever it is given.